### ISSUE TITLE
Merge pattern library into platform-client

### ISSUE THREAD

**Issue by rjmackay** [ushahidi/platform-client#943]
## Why?

The pattern library has become hard to maintain - with packages sometimes breaking and sometimes lagging behind platform-client. We want to encourage developers (and OSS users) to make changes to the pattern library when needed without the friction of moving form one system to another

## How

The initial shift will be mostly a move, rather than a refactor

- [ ] Move existing pattern library markup to platform-client/pattern-library
- [ ] Move existing pattern library sass into platform-client/sass
- [ ] Move font and img assets to platform-client/assets ? (not sure on naming)
- [ ] Move JS to platform-client/pattern-library/js ?
- [ ] Create a simple webpack config to build and develop on the PL.
    - This should render pattern library templates, CSS and JS
- [ ] Migrate bourbon and neat dependencies to npm rather than bower
- [ ] Remove handlebars templating *or* render templates at build time, not in the browser
- [ ] Set up auto deployment of built PL to preview.ushahidi.com again

**Comment by Angamanga** [comment]
@rjmackay Does this mean just moving the handlebars js just as is? Moving to angular is a totally different issue maybe?

**Comment by Angamanga** [comment]
@rjmackay Actually nevermind, when I think about it, since we are considering changing js-framework its better to wait on that before doing anything...

**Comment by rjmackay** [comment]
I'm not sure the pattern library should rely on angular or other JS frameworks. Keeping it as simple static HTML is more useful - nothing to break, its just a testbed for CSS.
Though maybe we need a testbed for UI components too - but they should probably be semi-separate.

**Comment by middle8media** [comment]
I agree with all of this @rjmackay, especially removing Handlebars and the confusion between Modules/Blocks, etc... The main issue for me would be setting up the WebPack build and the auto deployment.

**Comment by middle8media** [comment]
So for now the plan is to keep the PL to Platform flow as is, but drastically simplify the PL, making it easier for devs to jump in and revise/create components. Basically this means removing unnecessary dependencies like Handlebars. It would be helpful if @crcommons @Angamanga @justinscherer can chime in real quick to remind me of the biggest issues and blockers. This will help determine how it needs to be simplified.

**Comment by Angamanga** [comment]
@middle8media It looks good to me. I would like to add that while/if we look at the modules, blocks and fragments, we should also go through each section and see if the html differ from the layouts (I think they do in some cases) and correct this. We should add parts from the layouts that are not present in the modules/blocks/fragments and remove those who are not used anymore. To clean it up a bit.

**Comment by justinscherer** [comment]
@Erioldoesdesign suggested an [interface inventory](http://bradfrost.com/blog/post/interface-inventory/) which we should probably do anyways. This might catch the misalignments @Angamanga

**Comment by middle8media** [comment]
@rjmackay I'd like to chat more about these:

> Remove handlebars templating or render templates at build time, not in the browser
>Set up auto deployment of built PL to preview.ushahidi.com again

**Comment by rjmackay** [comment]
> Set up auto deployment of built PL to preview.ushahidi.com again

It should still be auto deploying through Travis-CI. The latest build seems to have passed and deploy: https://travis-ci.org/ushahidi/platform-pattern-library/jobs/334892948
Are you seeing issues?

**Comment by middle8media** [comment]
@rjmackay all good.

**Comment by sethburtonhall** [comment]
Can we close this? I think this would be covered via the React CL
